### PR TITLE
#6860 switch to new attachments interface

### PR DESCRIPTION
**Pull request by wp0pw**
Use new shape for attachments in attributes.
Attachments are now top level attributes
```
"photo1": {
       "arc_attribute_type": "arc_attachment",
       "arc_blob_identity": "/archivist/v1/blobs/234234",
       "arc_file_name": "The photo.jpeg",
       "arc_blob_hash_alg": "sha256",
       "arc_blob_hash_value": "234532453453245",
       "something_else": "my favourite",
}
```

### CODE REVIEW

File: archivist/assets.py [hunk 1]
```diff
@@ -250,9 +251,10 @@ def create_if_not_exists(
 
         # any attachments ?
         if attachments is not None:
-            data["attributes"]["arc_attachments"] = [
-                self._archivist.attachments.create(a) for a in attachments
-            ]
+            for a in attachments:
+                data["attributes"][
+                    a["attachment"]
```
Comment (henry739) [inline archivist/assets.py:256]: How do we know that the key `attachment` will be present? In the YAML example further up in the function, it suggests that filename will be there - but I see no mention of this key.

File: archivist/attachments.py [hunk 2]
```diff
@@ -126,11 +130,14 @@ def create(self, data: dict[str, Any]) -> dict[str, Any]:  # pragma: no cover
             attachment = self.upload(fd, mtype=data.get("content_type"))
 
         result = {
-            "arc_attachment_identity": attachment["identity"],
-            "arc_hash_alg": attachment["hash"]["alg"],
-            "arc_hash_value": attachment["hash"]["value"],
+            "arc_blob_identity": attachment["identity"],
```
Comment (henry739) [inline archivist/attachments.py:133]: Do we need to add "arc_attribute_type": "arc_attachment" here?

File: unittests/testassetsconstants.py [hunk 3]
```diff
@@ -20,20 +20,20 @@
...
     "arc_display_name": "arc_primary_image",
-    "arc_attachment_identity": "blobs/87b1a84c-1c6f-442b-923e-a97516f4d275",
-    "arc_hash_alg": "SHA256",
-    "arc_hash_value": "246c316e2cd6971ce5c83a3e61f9880fa6e2f14ae2976ee03500eb282fd03a60",
+    "arc_blob_identity": "blobs/87b1a84c-1c6f-442b-923e-a97516f4d275",
+    "arc_blob_hash_alg": "SHA256",
+    "arc_blob_hash_value": "246c316e2cd6971ce5c83a3e61f9880fa6e2f14ae2976ee03500eb282fd03a60",
 }
 SECONDARY_IMAGE = {
     "arc_display_name": "arc_secondary_image",
```
Comment (henry739) [inline unittests/testassetsconstants.py:28]: Think all of these need the attribute type

**Review by eccles — COMMENTED** [review]
Have we checked the taskrunner and the functests ?
Do the functests still work ?

Otherwise LGTM

**Review by henry739 — APPROVED** [review]
LGTM, but needs signoff from @eccles or @JAG-UK

**Pull request merged**